Sources/ISO/TSReader.swift

```diff
@@ -103,12 +103,14 @@ public class TSReader {
         defer {
             packetizedElementaryStreams[id] = nil
         }
-        if formatDescriptions[id] == nil {
-            formatDescriptions[id] = makeFormatDescription(data, pes: pes)
-            if let formatDescription = formatDescriptions[id] {
-                delegate?.reader(self, id: id, didRead: formatDescription)
+       
+        if let newFormat = makeFormatDescription(data, pes: pes) {
+            if formatDescriptions[id]?.dimensions.width != newFormat.dimensions.width || formatDescriptions[id]?.dimensions.height != newFormat.dimensions.height {
+                formatDescriptions[id] = newFormat
+                delegate?.reader(self, id: id, didRead: newFormat)
             }
```

Comment:
I believe it's more appropriate to handle format changes when parsing PMT rather than handling them here.
https://github.com/shogo4405/HaishinKit.swift/pull/1429/files#diff-ac8556870852ed52bd08e5610c2ecd5802d9bc3e3574a40ff39cc37bb2b8dc3aR33-R36